### PR TITLE
feat/reduce_calls_for_cookies

### PR DESCRIPTION
**Pull request by aarmoa**
- Changed the logic that processes cookies, to try and get the new cookie from any gRPC call response.
- Removed the extra initial request to get the cookie
- Removed the Kubernetes specific cookie assistant, and created a generic expiring cookie assistant.

https://injective-labs.atlassian.net/browse/CHAIN-20?atlOrigin=eyJpIjoiZDFlNmFlOWJjNjdkNGI3NjgwN2QwOTcyZDI3MDA2NzUiLCJwIjoiaiJ9

## Summary by CodeRabbit

- **Refactor**
    - Simplified cookies management logic for better user experience.
    - Enhanced code readability by centralizing metadata retrieval logic.
    - Improved handling of cookies and metadata for structured management.
- **Tests**
    - Streamlined test setup for various APIs, improving code readability and maintainability.

### CODE REVIEW

File: pyinjective/core/network.py [hunk 1]
```diff
@@ -1,171 +1,103 @@
...
-        self._exchange_cookie = cookie_info
+        if self._cookie is not None:
+            cookie = SimpleCookie()
+            cookie.load(self._cookie)
+            cookie_map = cookie
 
-    def _check_chain_cookie_expiration(self):
-        if self._is_cookie_expired(cookie_data=self._chain_cookie):
-            self._chain_cookie = None
+            for key in self._expiration_time_keys_sequence[:-1]:
```
Comment (keithbaum) [inline pyinjective/core/network.py:84]: Why are you skipping last item in this iteration?

File: pyinjective/core/network.py [hunk 2]
```diff
@@ -1,171 +1,103 @@
...
+                cookie_map = cookie.get(key, {})
 
-    def _check_exchange_cookie_expiration(self):
-        if self._is_cookie_expired(cookie_data=self._exchange_cookie):
-            self._exchange_cookie = None
+            expiration_data: Optional[str] = cookie_map.get(self._expiration_time_keys_sequence[-1], None)
+            if expiration_data is not None:
```
Comment (keithbaum) [inline pyinjective/core/network.py:88]: So if key is missing in the cookie_map then it's not expired. Is this confirmed to be this way by default?

**Review by keithbaum — COMMENTED** [review]
Understood the applied change and agree with implementation. Thanks

**Review by keithbaum — APPROVED** [review]

File: pyinjective/core/network.py (hunk 2 above)
Comment (aarmoa) [reply pyinjective/core/network.py:88]: Yes, correct. I have tested it with the current bare metal load balanced servers, and the k8s load balanced servers. If cookie is not renewed, then the response from the server does not include the info

File: pyinjective/core/network.py (hunk 1 above)
Comment (aarmoa) [reply pyinjective/core/network.py:84]: The variable `_expiration_time_keys_sequence` has all the keys that should be used to access the cookie in sequence (the cookie can be in a nested dictionary, to explain it better). So we use the list of keys without the last element just to keep going into the nested dictionaries (like a Mamushka of dictionaries), and the last element to access the cookie value

**Pull request merged**